### PR TITLE
DEV 870 - Interstitial form validation failure when changing credit card payment method from contributor portal

### PR DESCRIPTION
**Pull request by benjaminEwhite**
What's this PR do?
Will not allow user to submit payment update if there are any errors or partially filled payment details.

Why are we doing this? How does it help us?
Fixing this but improves user experience.

How should this be manually tested?
Goto -> `/contributor/contributions/`-> click on payment method (card number) -> provide the card details.
`Update payment method` should not be enabled if the payment details are not completely provided.

Have automated unit tests been added?
Yes

How should this change be communicated to end users?
N/A

Are there any smells or added technical debt to note?
No

Has this been documented? If so, where?
N/A

What are the relevant tickets?
(Note: Please use syntax [JIRA-123] to link any relevant tickets)
[DEV-870]

Do any changes need to be made before deployment to production (adding environment variables, for example)?
No

Are there next steps? If so, what? Have tickets been opened for them?
(Note: Please use syntax [JIRA-123] to link any relevant tickets)

[DEV-870]: https://news-revenue-hub.atlassian.net/browse/DEV-870?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: spa/cypress/integration/08 - contributor.spec.js [hunk 1]
```diff
@@ -141,3 +141,83 @@ describe('Contributor portal', () => {
     });
   });
 });
+
+describe('Update recurring contribution modal', () => {
+  beforeEach(() => {
+    // "Log in" to contributor dash
+    cy.intercept({ method: 'POST', url: getEndpoint(VERIFY_TOKEN) }, { fixture: 'user/valid-contributor-1.json' }).as(
```
Comment (benjaminEwhite) [inline spa/cypress/integration/08 - contributor.spec.js:148]: There's a custom Cypress command you should use for all logins. you can do `cy.login('user/valid-contributor-1.json')`.  In addition to setting up the intercept, that command also clicks the login button (which is maybe what's missing here).

File: spa/cypress/integration/08 - contributor.spec.js [hunk 2]
```diff
@@ -141,3 +141,83 @@ describe('Contributor portal', () => {
...
+    cy.interceptPaginatedDonations();
+    cy.visit(CONTRIBUTOR_VERIFY);
+    cy.wait(['@login', '@getDonations']);
+    cy.get('td[data-testcolumnaccessor="interval"]')
+      .contains('Monthly')
+      .closest('tr')
+      .find('[data-testid="payment-method"]')
+      .first()
+      .click()
+      .wait(1000);
```
Comment (benjaminEwhite) [inline spa/cypress/integration/08 - contributor.spec.js:160]: .wait + an integer value is almost always a code smell in Cypress (vs. .wait + aliases, which is a great pattern).

File: spa/cypress/integration/08 - contributor.spec.js [hunk 3]
```diff
@@ -141,3 +141,83 @@ describe('Contributor portal', () => {
...
+  });
+
+  afterEach(() => {
+    cy.getByTestId('close-modal').click();
+  });
```
Comment (benjaminEwhite) [inline spa/cypress/integration/08 - contributor.spec.js:165]: The modal will be in its default closed state at the beginning of each test anyways.

File: spa/cypress/support/commands.js [hunk 4]
```diff
@@ -110,3 +110,13 @@ Cypress.Commands.add('editElement', (elementType) => {
       cy.getByTestId('pencil-button').click({ force: true });
     });
 });
+
+Cypress.Commands.add('getStripeCardElement', (elementName) => {
+  return cy
+    .wait(1000)
```
Comment (benjaminEwhite) [inline spa/cypress/support/commands.js:116]: This `.wait` shouldn't be here.

File: spa/cypress/integration/08 - contributor.spec.js (hunk 2 above)
Comment (harsha09) [reply spa/cypress/integration/08 - contributor.spec.js:160]: Yeah, I have put this for testing purpose in git workflow

File: spa/cypress/support/commands.js (hunk 4 above)
Comment (harsha09) [reply spa/cypress/support/commands.js:116]: This is for testing purpose only

File: spa/cypress/integration/08 - contributor.spec.js (hunk 1 above)
Comment (harsha09) [reply spa/cypress/integration/08 - contributor.spec.js:148]: It doesn't actually login, it requires magic link to be able to view this page

**Comment by gurjittoptal** [comment]
looks good to me.

**Review by gurjittoptal — APPROVED** [review]

**Pull request merged**